platform/engineering/request-for-comment/2022-05-05-mobile-route-naming-conventions.md

```diff
@@ -0,0 +1,201 @@
+# RFC: Mobile Routes naming conventions
+
+- Date: 2022-05-05
+- Related Issue: https://app.zenhub.com/workspaces/va-mobile---backend-61d75d03a7505500147ec7f6/issues/department-of-veterans-affairs/va-mobile-app/2937
+
+## Background
+Mobile routes currently have some inconsistencies in their naming conventions, such as:
+- Sometimes using snake_case over kebab-case (maintenance_windows, message_drafts)
+- :ids are after the resource action rather than before (canceling appointments)
+- Not nesting under parent products (the 'health' and 'benefits' app)
+
+## Motivation
+Keeping naming consistent will make using the Mobile API much easier to use and is generally a good practice for maintainability. 
+
+## Design
+Version all endpoint that do not meet the naming convention as follows:
+```
+namespace :health do
+  get '/appointment-requests/:appointment_request_id/messages', to: 'appointment_request_messages#index'
+  get '/appointments', to: 'appointments#index'
+  put '/appointments/:id/cancel', to: 'appointments#cancel'
+  get '/appointments/community-care/:service_type/eligibility', to: 'community_care_eligibility#show'
+  get '/appointments/va/eligibility', to: 'veterans_affairs_eligibility#show'
+  get '/appointments/facility/eligibility', to: 'facility_eligibility#index'
+  post '/appointment', to: 'appointments#create'
+  get '/community-care-providers', to: 'community_care_providers#index'
```

Comment:
Maybe `/appointments/community-care-providers` ?

Reply:
Andrew and I discussed that when I created the endpoint. Our rationale for not doing it is that there could be some theoretical reason that we might need to use providers for a reason not directly linked to appointments (like showing a map of locations or something). I think that's very unlikely and I'm happy with it either way. It sounds like we're trying to do a better job of organizing our module, which is a good thing.

Reply:
yeah, I liked the idea at the time but now for the sake of simplicity it does seem cleaner put appointments in the url. Maybe we just keep it as a separate controller in case of the unlikely event that we need to implement more actions for cc providers? probably should do the same with the facility info endpoint as well then.